File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/elementcache/ElementCacheBehaviour.java
package com.gigigo.orchextra.core.domain.entities.elementcache;

import java.io.Serializable;

public enum ElementCacheBehaviour implements Serializable {
  CLICK("click"),
  SWIPE("swipe"),
  NONE("");

  private final String behaviour;

  ElementCacheBehaviour(String behaviour) {
    this.behaviour = behaviour;
  }

  public String getBehaviour() {
    return behaviour;
  }

  public static ElementCacheBehaviour convertStringToEnum(String behaviour) {
    ElementCacheBehaviour[] values = ElementCacheBehaviour.values();
    for (ElementCacheBehaviour value : values) {
      if (value.getBehaviour().equals(behaviour)) {
        return value;
      }
    }
    return NONE;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/elementcache/ElementCacheBehaviour.kt
package com.gigigo.orchextra.core.domain.entities.elementcache

import java.io.Serializable

enum class ElementCacheBehaviour constructor(val cacheBehaviour: String) : Serializable {
  CLICK("click"),
  SWIPE("swipe"),
  NONE("");

  companion object {

    fun convertStringToEnum(behaviour: String?): ElementCacheBehaviour {
      val values = ElementCacheBehaviour.values()
      for (value in values) {
        if (value.cacheBehaviour.equals(behaviour, true)) {
          return value
        }
      }
      return NONE
    }
  }
}